[BH-1077] Delete taxonomies

apmatthews opened the pull request:
## Description

<!--
Include a summary of the change and some contextual information.
-->

- Enables deletion of taxonomies.
- Adds a helper for acceptance tests to simplify initial taxonomy creation.

<!--
Provide a link to the JIRA ticket (if any) for issue tracking purposes
-->

https://wpengine.atlassian.net/browse/BH-1077

## Testing

<!--
Describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Also list any relevant details for your test configuration such as how to test the changes locally or in staging.
-->

Integration tests are included to verify that the DELETE route:
1. Is registered
2. Handles valid requests and returns a proper response.
3. Handles invalid requests (i.e. taxonomies not created with ACM) and returns a proper response.

The included acceptance test verifies that deletion works as expected in the UI.

To test manually:

1. Create at least one model to assign your taxonomy to.
2. Click “View Taxonomies" from the main model page or visit `/wp-admin/?page=atlas-content-modeler&view=taxonomies`
3. Add one or more taxonomies, which should appear in the table.
4. Click the "..." options menu for one of your listed taxonomies and then click “Delete”.
5. You should see a delete confirmation modal.
6. Click "Delete" in the modal.
7. See that the taxonomy is removed from the taxonomy list table (immediate) as well as from the associated model (currently requires a page refresh).

## Screenshots

<!--
If this is a visual change include relevant screenshots about the behavior of the application before and after this change.
-->
n/a

## Documentation Changes

<!--
List corresponding changes to the documentation and updated wiki pages.
-->
n/a

## Dependant PRs

<!--
List any dependent PR's that are awaiting review. And in order to have these dependencies listed as part of the checks section, use the following syntax:

Depends on #1234
-->
May conflict or overlap with #190.

apmatthews committed:
feat: add delete taxonomy functionality

apmatthews committed:
refactor: simplify parameter retrieval

apmatthews committed:
refactor: improve error response

apmatthews committed:
test: add tests for delete REST routes

apmatthews committed:
refactor: use taxonomy plural name in error message

apmatthews committed:
chore: convert commented code to TODO

apmatthews committed:
test: add acceptance test for taxonomy deletion

apmatthews committed:
chore: ensure types is an array in haveTaxonomy helper

nickcernis reviewed (approved):
Looking great! Let's merge this first and I'll rebase/adjust the edit PR later.

nickcernis committed:
chore: amend spacing to trigger CircleCI

Circle had downtime yesterday and doesn't seem
to have found this new branch.

This is a spacing change purely to try to trigger
Circle workflows.

nickcernis commented:
@apmatthews I pushed an (unneeded but low-impact) docs spacing change in 2934b2a to trigger Circle workflows after it failed to find your branch here during its downtime yesterday.

It worked and workflows have passed. 🎉

I'm happy to merge as-is if you are. Feel free to revert my docs change if you'd like to. (It wasn't required for approval here; it was just a less destructive way to get Circle to pick up the ghost branch than a force-push or branch name change.)

apmatthews committed:
style: convert spaces to tabs
